tegel/src/components/divider/divider.tsx

```diff
@@ -0,0 +1,64 @@
+import { Component, Prop, Host, h } from '@stencil/core';
+
+@Component({
+  tag: 'sdds-divider',
+  styleUrl: 'divider-component.scss',
+  shadow: true,
+})
+export class Divider {
+  /** Direction of the Divider, horizontal if not specified. */
+  @Prop() direction: 'horizontal' | 'vertical' = 'horizontal';
+
+  /** Sets the divider to light version */
+  @Prop() light: boolean = false;
+
+  /** Sets the divider to dark version */
+  @Prop() dark: boolean = false;
+
+  /** Width prop, applicable to the horizontal version */
+  @Prop() width?: string;
+
+  /** Height prop, applicable to the vertical version */
+  @Prop() height?: string;
+
+  render() {
+    let colorClass = '';
+
+    if (this.light) {
+      colorClass = 'sdds-theme-light';
+    } else if (this.dark) {
+      colorClass = 'sdds-theme-dark';
+    }
+
+    const separatorProps = {
+      'role': 'separator',
```

Comment:
Since it's always gonna be a separator, could we just set this on the host element in the return straight away instead. I think this is a nice way of building up to the return, it's just different to how we have done it in other component. We have tried to use stencils lifecycle to to add/change/remove properties based on other properties or other stuff that might impact the component.

Reply:
I see, so something like this instead:
```suggestion
      return (
          <Host
            role="separator"
            aria-orientation={this.direction === 'vertical' ? 'vertical' : undefined}
            class={`sdds-divider ${this.direction} ${colorClass}`}
            style={{
              width: this.direction === 'horizontal' && this.width,
              height: this.direction === 'vertical' && this.height,
            }}
          >
            {label && <span class="sdds-divider-label" aria-hidden>{label}</span>}
            {this.direction === 'horizontal' ? <hr style={{width: '1px', height: '1px'}} /> : <div style={{width: '1px', height: '1px'}} />}
          </Host>
     );
```

But without the width height props maybe and the label then?

Reply:
Yeah exactly, nice!